crates/sui-framework/packages/dwallet-system/sources/dwallet.move

```diff
@@ -472,13 +473,14 @@ module dwallet_system::dwallet {
 
     /// Register an encryption key to encrypt a user share.
     /// The key is saved as an immutable object.
-    public fun register_encryption_key(key: vector<u8>, scheme: u8, ctx: &mut TxContext) {
+    public fun register_encryption_key(key: vector<u8>, signed_key: vector<u8>, scheme: u8, ctx: &mut TxContext) {
         assert!(is_valid_encryption_key_scheme(scheme), EInvalidEncryptionKeyScheme);
         let encryption_key = EncryptionKey {
```

Comment:
You need to add signature verification here in the blockchain Move side

Reply:
yes this is in addition. We don't want the transaction to succeed if the user provides invalid inputs